controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -259,6 +268,42 @@ func (r *BareMetalHostReconciler) Reconcile(request ctrl.Request) (result ctrl.R
 	return
 }
 
+// Consume inspect.metal3.io/hardwaredetails when either
+// inspect.metal3.io=disabled or there are no existing HardwareDetails
+func (r *BareMetalHostReconciler) updateHardwareDetails(request ctrl.Request, host *metal3v1alpha1.BareMetalHost) (bool, error) {
+	updated := false
+	if host.Status.HardwareDetails == nil || inspectionDisabled(host) {
+		objHardwareDetails, err := r.getHardwareDetailsFromAnnotation(host)
+		if err != nil {
+			return updated, errors.Wrap(err, "Error getting HardwareDetails from annotation")
+		}
+		if objHardwareDetails != nil {
+			r.publishEvent(request, host.NewEvent("UpdateHardwareDetails", "Setting HardwareDetails from annotation"))
```

Comment:
Wouldn't be safer to postpone events publishing just after a successful `r.saveHostStatus`?

Reply:
Good point, I moved the events to after the status is stored in both cases.